kedro-datasets/kedro_datasets/pandas/sql_dataset.py

```diff
@@ -275,6 +283,25 @@ def _exists(self) -> bool:
         schema = self._load_args.get("schema", None)
         return insp.has_table(self._load_args["table_name"], schema)
 
+    def preview(self, nrows: int = 5) -> TablePreview:
+        """
+        Generate a preview of the dataset with a specified number of rows.
+
+        Args:
+            nrows: The number of rows to include in the preview. Defaults to 5.
+
+        Returns:
+            dict: A dictionary containing the data in a split format.
+        """
+
+        table_name = _sanitise_table_name(self._load_args["table_name"])
+
+        sql_query = f"SELECT * FROM {table_name} LIMIT {nrows}"  # nosec
```

Comment:
This allow SQL injection and thus the security error? Do we have better alternatives?

Reply:
we did it like this in Kedro-viz - https://github.com/kedro-org/kedro-viz/blob/main/package/kedro_viz/data_access/repositories/runs.py#L51

Reply:
we could also do without creating a model - https://stackoverflow.com/questions/49005812/python-sqlalchemy-access-huge-db-data-without-creating-models

Reply:
Addressed the issue by using SQLAlchemy library instead.